crates/anvil/src/eth/backend/mem/mod.rs

```diff
@@ -2626,6 +2635,126 @@ impl Backend {
+
+        let parent = self.get_block_by_hash(block.header.parent_hash).unwrap();
+        let parent_number = parent.header.number();
+
+        let prior_txs = &block.transactions[..index];
+        let pool_txs: Vec<Arc<PoolTransaction>> = prior_txs
+            .iter()
+            .map(|tx| {
+                let typed_tx: TypedTransaction = tx.clone().into();
+                let pending_tx = PendingTransaction::new(typed_tx).unwrap();
+                Arc::new(PoolTransaction {
+                    pending_transaction: pending_tx,
+                    requires: vec![],
+                    provides: vec![],
+                    priority: crate::eth::pool::transactions::TransactionPriority(0),
+                })
+            })
+            .collect();
+
+        let db = self.db.read().await;
+        let result = {
+            self.with_database_at(
+                Some(BlockRequest::Number(parent_number)),
+                |_state_db, block_env| {
+                    let mut cache_db = CacheDB::new(Box::new(&*db));
+
+                    let executor = TransactionExecutor {
+                        db: &mut cache_db,
+                        validator: self,
+                        pending: pool_txs.into_iter(),
+                        block_env: block_env.clone(),
+                        cfg_env: self.next_env().evm_env.cfg_env,
+                        parent_hash: block.header.parent_hash,
+                        gas_used: 0,
+                        blob_gas_used: 0,
+                        enable_steps_tracing: self.enable_steps_tracing,
+                        print_logs: self.print_logs,
+                        print_traces: self.print_traces,
+                        call_trace_decoder: self.call_trace_decoder.clone(),
+                        precompile_factory: self.precompile_factory.clone(),
+                        odyssey: self.odyssey,
+                        optimism: self.is_optimism(),
+                        blob_params: self.blob_params(),
+                    };
+
+                    let _ = executor.execute();
+
+                    let built_tx = transaction_build(
+                        Some(info.transaction_hash),
+                        block.transactions[index].clone(),
+                        Some(&block),
+                        Some(info),
+                        block.header.base_fee_per_gas,
+                    );
+
+                    let request = TransactionRequest::from_transaction(built_tx);
+                    let call_env = self.build_call_env(
+                        WithOtherFields::new(request.clone()),
+                        FeeDetails::new(
+                            request.gas_price,
+                            request.max_fee_per_gas,
+                            request.max_priority_fee_per_gas,
+                            request.max_fee_per_blob_gas,
+                        )
+                        .unwrap(),
```

Comment:
this we'd to handle differently:

https://github.com/foundry-rs/foundry/blob/391f37f063a625af53a2bdb7a5af37fbb160c750/crates/anvil/core/src/eth/transaction/mod.rs#L396-L400